comfy/model_management.py

```diff
@@ -303,9 +310,13 @@ def model_load(self, lowvram_model_memory=0, force_patch_weights=False):
             self.model.unpatch_model(self.model.offload_device)
             self.model_unload()
             raise e
+        
+        if cpu_state == CPUState.CPU:
+            torch.set_autocast_cpu_enabled(use_cpu_autocast())
 
-        if is_intel_xpu() and not args.disable_ipex_optimize:
-            self.real_model = ipex.optimize(self.real_model.eval(), graph_mode=True, concat_linear=True)
+        if (is_cpu_with_ipex() or is_intel_xpu()) and not args.disable_ipex_optimize:
+            ipex_dtype = torch.bfloat16 if cpu_state == CPUState.CPU and cpu_has_fast_bf16() else None
+            self.real_model = ipex.optimize(self.real_model.eval(), dtype=ipex_dtype, graph_mode=True, concat_linear=True)
```

Comment:
I think you want to consider turning on auto_kernel_selection here as an extra parameter to the optimize call. I never called optimize with it because it was listed as a CPU only optimization but it is one of the few flags that doesn't get turned on by default with the O1 optimize flag in torch.xpu.optimize. So you get the following for this line:
`self.real_model = ipex.optimize(self.real_model.eval(), dtype=ipex_dtype, graph_mode=True, auto_kernel_selection=True, concat_linear=True)`

Reply:
Checked `auto_kernel_selection` - it makes no changes, so I don't use it (ipex documentation suggest to use default flags when possible).

For some reason `torch.compile(model, backend="ipex")` also makes no difference on CPU (Intel/AMD) or XPU, even though ipex documentation recommends it.

Reply:
Alright, I guess. I turned on concat linear and graph mode which weren't in O1 by default because they helped when turned on. Also, torch.compile doesn't help because ipex.optimize with graph mode turns on effectively gives most of the speedups that torch.compile does here. I had written a patch for torch.compile a long time ago but it never gave much speed improvements and wasn't put in the right places to be worth comfy's time to merge.